.github/actions/sign-attestation/src/main.ts

```diff
@@ -0,0 +1,39 @@
+import * as core from "@actions/core";
+import * as fs from "fs";
+import * as sigstore from "sigstore";
+
+const signOptions = {
+  oidcClientID: "sigstore",
+  oidcIssuer: "https://oauth2.sigstore.dev/auth",
+  rekorBaseURL: sigstore.sigstore.DEFAULT_REKOR_BASE_URL,
+};
+
+async function run(): Promise<void> {
+  try {
+    // `predicate` input defined in action metadata file
+    const attestation = core.getInput("attestation");
```

Comment:
I think we'll do the directory traversal verification in the caller. But adding it here in addition / peace of mind would probably be a good thing.